Modules/_ssl.c

```diff
@@ -3402,6 +3432,91 @@ _ssl__SSLContext_get_ciphers_impl(PySSLContext *self)
 
 }
 
+/*[clinic input]
+@critical_section
+_ssl._SSLContext.set_groups
+    grouplist: str
```

Comment:
Would it make sense to actually support a list of strings and then we join the list into a single string and pass it to OpenSSL? Maybe expose it as a pure Python method as well? And I suggest we call this method set_groups_list() and the pure python one set_groups(). (since get_groups() returns a list and not a string I think it makes sense). Tell me what you think.

Reply:
Actually, to support the full functionality, a string is required here. While the string can be a single group or a colon-separated list of groups in the simplest cases, the string format actually supports a much more complex structure. It's actually a slash-separate list of "group tuples" representing different priority levels. Each group tuple is then a colon-separated list of groups which are equal in priority. There are other modifiers as well in front of group names (minus, question mark, asterisk) as well. This is all documented in the link I included in the doc (https://docs.openssl.org/master/man3/SSL_CTX_set1_groups_list).